components/engine_rocks/src/misc.rs

```diff
@@ -86,18 +88,23 @@ impl RocksEngine {
             for key in data.iter() {
                 wb.delete_cf(cf, key)?;
                 if wb.count() >= Self::WRITE_BATCH_MAX_KEYS {
-                    wb.write()?;
+                    wb.write_opt(wopts)?;
                     wb.clear();
                 }
             }
             if wb.count() > 0 {
-                wb.write()?;
+                wb.write_opt(wopts)?;
             }
         }
         Ok(())
     }
 
-    fn delete_all_in_range_cf_by_key(&self, cf: &str, range: &Range<'_>) -> Result<()> {
+    fn delete_all_in_range_cf_by_key(
```

Comment:
Could you remove `self.should_skip(off, index)` in `Apply::apply_delete_range`? Because delete range no longer be traced by ApplyTrace, without WAL, it may leaks some keys after restart.

Reply:
That would be incorrect, RocksDB write is not idempotent.

> Because delete range no longer be traced by ApplyTrace, without WAL, it may leaks some keys after restart.

How exactly? If memtable isn't flushed after delete range is applied, the log_recovery won't skip to re-apply it after restart.